precompiles/staking/types.go

```diff
@@ -49,6 +56,110 @@ type EventCancelUnbonding struct {
+	MaxRate       *big.Int "json:\"maxRate\""
+	MaxChangeRate *big.Int "json:\"maxChangeRate\""
+}
+
+// NewMsgCreateValidator creates a new MsgCreateValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgCreateValidator, common.Address, error) {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	descriptionJson, err := json.Marshal(args[0])
+	if err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to marshal description to json: %v", err)
+	}
+	var description Description
+	if err := json.Unmarshal(descriptionJson, &description); err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to unmarshal json to description: %v", err)
+	}
+
+	commissionJson, err := json.Marshal(args[1])
+	if err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to marshal commission to json: %v", err)
+	}
+	var commission Commission
+	if err := json.Unmarshal(commissionJson, &commission); err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to unmarshal json to commission: %v", err)
+	}
+
+	minSelfDelegation, ok := args[2].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidAmount, args[2])
+	}
+
+	delegatorAddress, ok := args[3].(common.Address)
+	if !ok || delegatorAddress == (common.Address{}) {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidDelegator, args[3])
+	}
+
+	validatorAddress, ok := args[4].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "validatorAddress", "string", args[4])
+	}
+
+	// use cli `evmosd tendermint show-validator` get pubkey
+	pubkeyBase64Str, ok := args[5].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "pubkey", "string", args[5])
+	}
+	pubkeyBytes, err := base64.StdEncoding.DecodeString(pubkeyBase64Str)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+	var ed25519pk cryptotypes.PubKey = &ed25519.PubKey{Key: pubkeyBytes}
+	pubkey, err := codectypes.NewAnyWithValue(ed25519pk)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+
+	value, ok := args[6].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidAmount, args[6])
+	}
+
+	msg := &stakingtypes.MsgCreateValidator{
+		Description: stakingtypes.Description{
+			Moniker:         description.Moniker,
+			Identity:        description.Identity,
+			Website:         description.Website,
+			SecurityContact: description.SecurityContact,
+			Details:         description.Details,
+		},
+		Commission: stakingtypes.CommissionRates{
+			Rate:          sdk.NewDecFromBigIntWithPrec(commission.Rate, math.LegacyPrecision),
+			MaxRate:       sdk.NewDecFromBigIntWithPrec(commission.MaxRate, math.LegacyPrecision),
+			MaxChangeRate: sdk.NewDecFromBigIntWithPrec(commission.MaxChangeRate, math.LegacyPrecision),
```

Comment:
pretty sure we have a precision value that is defined by Evmos. Check the types or app.go

Reply:
By searching for the keyword **= 18** in evmos I only found BaseDenomUnit = 18. I think it is unreasonable to use this, so I replaced it with the **Precision** constant in the types in cosmos sdk.